demo-kubernetes-with-libstorage/README.md

```diff
@@ -0,0 +1,71 @@

Use the AWS console to verify that the volume was created and that it is attached to the VM

This will show the log entries related to the attachment:
`cat /tmp/kube-controller-manager.log | grep libstorage`

### Step 11: Create a PersistentVolumeClaim and utilize it

Create an AWS ebs volume:
`rexray volume create vol-0002 --size=1`

Create a Kubernetes persistent volume referencing it:

Create the file pv.yml with this content:

+```
kind: PersistentVolume
apiVersion: v1
metadata:
  name: pv-0001
spec:
  capacity:
    storage: 1Gi
  accessModes:
    - ReadWriteOnce
  libStorage:
    host: http://127.0.0.1:7979
    service: ebs
    volumeName: vol-0002
```
+
`cluster/kubectl.sh create -f pv.yml`

Create a Kubernetes persistent volume claim:

Create the file pvc.yml with this content:

+```
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: pvc-0001
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 1Gi
```
+
`cluster/kubectl.sh create -f pvc.yml`

Create a pod that uses the claim:

Create the file pod-pvc.yml with this content:

+```
kind: Pod
apiVersion: v1
metadata:
  name: podpvc-0001
spec:
  containers:
    - name: pod0002-container
      image: gcr.io/google_containers/test-webserver
      volumeMounts:
      - mountPath: /test
        name: test-data
  volumes:
    - name: test-data
      persistentVolumeClaim:
        claimName: pvc-0002
```
+
`cluster/kubectl.sh create –f pod-pvc.yml`

### Step 12: Utilize dynamic provisioning by storage class

Create a storage class:

Create the file sc.yml with this content:

+```
apiVersion: storage.k8s.io/v1beta1
kind: StorageClass
metadata:
  name: sc-0001
provisioner: kubernetes.io/libstorage
parameters:
  host: http://127.0.0.1:7979
  service: ebs
```
+
`cluster/kubectl.sh create –f sc.yml`

Create a persistent volume claim that utilizes the storage class:

Create the file sc-pvc.yml with this content:

+```
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: pvc-0002
  annotations:
      volume.beta.kubernetes.io/storage-class: sc-0001
spec:
  accessModes:
    - ReadWriteOnce
  resources:
    requests:
      storage: 1Gi
```
+
`cluster/kubectl.sh create –f sc-pvc.yml`

Create a pod that uses the persistent volume claim:

Create the file pod-sc-pvc.yml with this content:

+```
kind: Pod
apiVersion: v1
metadata:
  name: podscpvc-0001
spec:
  containers:
    - name: pod0003-container
      image: gcr.io/google_containers/test-webserver
      volumeMounts:
      - mountPath: /test
        name: test-data
  volumes:
    - name: test-data
      persistentVolumeClaim:
        claimName: pvc-0002
```
+
`cluster/kubectl.sh create –f pod-sc-pvc.yml`
+
+## Contribution
+
+Create a fork of the project into your own repository. Make all your necessary changes and create a pull request with a description on what was added or removed and details explaining the changes in lines of code. If approved, project owners will merge it.
+
+
+## Support
```

Comment:
Steve, `glide` is not really required (at least not for the Kubernetes codebase)